src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems;
+
+import com.chopshop166.chopshoplib.logging.LoggedSubsystem;
+import com.chopshop166.chopshoplib.pneumatics.IDSolenoid;
+import com.chopshop166.chopshoplib.pneumatics.MockDSolenoid;
+import com.chopshop166.chopshoplib.sensors.IColorSensor;
+
+import frc.robot.maps.subsystems.IntakeData;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class Intake extends LoggedSubsystem<IntakeData, IntakeData.Map> {
+
+    private IDSolenoid solenoid = new MockDSolenoid();
+    private final double GRAB_SPEED = 1;
+    private final double RELEASE_SPEED = -1;
+    private IColorSensor colorSensor;
```

Comment:
This also needs to be moved into the map. and the IntakeData should probably have a double for the distance and a color value.